pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -0,0 +1,162 @@
+	return sandboxName
+}
+
+// parseSandboxName unpacks a sandbox full name, returning the pod name, namespace and uid
+func parseSandboxName(name string) (string, string, string, error) {
+	podName, podNamespace, podUID, _, _, err := parseContainerName(name)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	return podName, podNamespace, podUID, nil
+}
+
+// buildContainerName creates a name which can be reversed to identify container name.
+// This function returns stable name, unique name and an unique id.
+func buildContainerName(pod *api.Pod, container *api.Container) (string, string, string) {
+	// kubelet uses hash to determine whether an existing container matches the desired spec.
+	containerName := container.Name + "." + strconv.FormatUint(kubecontainer.HashContainer(container), 16)
+	return buildKubeGenericName(pod, containerName)
+}
+
+// buildKubeGenericName creates a name which can be reversed to identify container/sandbox name.
+// This function returns stable name, unique name and an unique id.
+func buildKubeGenericName(pod *api.Pod, containerName string) (string, string, string) {
+	stableName := fmt.Sprintf("%s_%s_%s_%s_%s",
+		kubePrefix,
+		containerName,
+		pod.Name,
+		pod.Namespace,
+		string(pod.UID),
+	)
+	UID := fmt.Sprintf("%08x", rand.Uint32())
+	return stableName, fmt.Sprintf("%s_%s", stableName, UID), UID
+}
+
+// parseContainerName unpacks a container name, returning the pod name, namespace, UID and container name
+func parseContainerName(name string) (podName, podNamespace, podUID, containerName string, hash uint64, err error) {
+	parts := strings.Split(name, "_")
+	if len(parts) == 0 || parts[0] != kubePrefix {
+		err = fmt.Errorf("failed to parse container name %q into parts", name)
+		return "", "", "", "", 0, err
+	}
+	if len(parts) < 6 {
+		glog.Warningf("Found a container with the %q prefix, but too few fields (%d): %q", kubePrefix, len(parts), name)
+		err = fmt.Errorf("container name %q has fewer parts than expected %v", name, parts)
+		return "", "", "", "", 0, err
+	}
+
+	nameParts := strings.Split(parts[1], ".")
+	containerName = nameParts[0]
+	if len(nameParts) > 1 {
+		hash, err = strconv.ParseUint(nameParts[1], 16, 32)
+		if err != nil {
+			glog.Warningf("invalid container hash %q in container %q", nameParts[1], name)
+		}
+	}
+
+	return parts[2], parts[3], parts[4], containerName, hash, nil
+}
+
+// toRuntimeProtocol converts api.Protocol to runtimeApi.Protocol
+func toRuntimeProtocol(protocol api.Protocol) runtimeApi.Protocol {
+	switch protocol {
+	case api.ProtocolTCP:
+		return runtimeApi.Protocol_TCP
+	case api.ProtocolUDP:
+		return runtimeApi.Protocol_UDP
+	}
+
+	glog.Warningf("Unknown protocol %q: defaulting to TCP", protocol)
+	return runtimeApi.Protocol_TCP
+}
+
+// milliCPUToShares converts milliCPU to CPU shares
+func milliCPUToShares(milliCPU int64) int64 {
+	if milliCPU == 0 {
+		// Docker converts zero milliCPU to unset, which maps to kernel default
```

Comment:
I think I commented on this before, maybe it got lost during rebase. 
Remove the "Docker" reference in this comment.